third_party/2and3/sqlalchemy/sql/selectable.pyi

```diff
@@ -21,62 +27,226 @@ class HasSuffixes(object):
     def suffix_with(self, *expr, **kw): ...
 
 class FromClause(Selectable):
-    def count(self, functions, whereclause=None, **params): ...
-    def select(self, whereclause=None, **params): ...
-    def join(self, right, onclause=None, isouter: bool=False): ...
-    def outerjoin(self, right, onclause=None): ...
-    def alias(self, name=None, flat: bool=False): ...
+    __visit_name__ = ...  # type: str
+    named_with_column = ...  # type: bool
+    schema = ...  # type: Any
+    def count(self, functions, whereclause: Optional[Any] = ..., **params): ...
+    def select(self, whereclause: Optional[Any] = ..., **params): ...
+    def join(self, right, onclause: Optional[Any] = ..., isouter: bool = ..., full: bool = ...): ...
+    def outerjoin(self, right, onclause: Optional[Any] = ..., full: bool = ...): ...
+    def alias(self, name: Optional[Any] = ..., flat: bool = ...): ...
+    def lateral(self, name: Optional[Any] = ...): ...
+    def tablesample(self, sampling, name: Optional[Any] = ..., seed: Optional[Any] = ...): ...
     def is_derived_from(self, fromclause): ...
-    def _is_lexical_equivalent(self, other): ...
     def replace_selectable(self, sqlutil, old, alias): ...
     def correspond_on_equivalents(self, column, equivalents): ...
-    def corresponding_column(self, column, require_embedded: bool=False): ...
+    def corresponding_column(self, column, require_embedded: bool = ...): ...
     @property
     def description(self): ...
-    def _reset_exported(self): ...
-    @property
     def columns(self): ...
```

Comment:
Should stay as `@property`, sqlalchemy is using `@_memoized_property`, otherwise we can't access `X.columns[name]` for "function not being indexable".